File: src/HouseRobber.java
public class HouseRobber {
    public static int rob(int[] array) {
        if (array.length == 0) return 0;

        int[] memory = new int[array.length + 1];
        memory[1] = array[0];
        for (int index = 1 ; index < array.length ; index++) {
            memory[index + 1] = Math.max(memory[index], memory[index - 1] + array[index]);
        }
        return memory[memory.length - 1];
    }
}

File: python/house_robber.py
from typing import List


class Solution:
    def rob(self, nums: List[int]) -> int:
        a, b, max_amount = 0, nums[0], nums[0]
        for index in range(1, len(nums)):
            max_amount = max(a + nums[index], b)
            a, b = b, max_amount
        return max_amount
